### DISCUSSION TITLE
MenuButton control not yet fully integrated with the editor

### DISCUSSION THREAD

**Discussion by InfiniteProductions** [godotengine/godot-proposals#6572] [GUI]
Godot offer already plenty of controls to make GUI and there are signals we can use to manage user's actions, still, _MenuButton_ control is not fully integrated with the editor as we can mostly only add items and use a very limited number of signals, but most importantly, we're unable to make it fully usable right now as item selection signal is not available through the editor.

There are the two demos 2d and 3d physics_tests which use it and the item selection handling is made by code (they even create their own _OptionMenu_ extending _MenuButton_, I don't know if it is intended to be used this way).
Documentation does not cover the control very well either, at least the link to the demos could be added for those who are seeking for an example and are not aware of these.

(following of the Godot main project issue #75282)

**Comment by AThousandShips** [comment]
You can also achieve most of this functionality by simply adding a `PopupMenu` and controlling it from a button, slightly tricky to get the exact behavior of `MenuButton` but you can get access to all the functionality of the menu like connections, the only feature that's a bit tricky to get working like this is the `switch_on_hover`

One way to integrate this would be to relay the signals of the `PopupMenu` via the button, copying the signals from it and triggering them when the signals are triggered in the popup, but it's also very simple just to connect the signals via script

**Comment by AThousandShips** [comment]
(To add a link to this issue you need to write godotengine/godot#75282)
